 
 <robot name = "my_robot">
 <ros2_control name="imu_sensor" type="sensor">
  <hardware>
    <plugin>imu_hardware_interface/HwtImuInterface</plugin>
    <param name="port">/dev/ttyUSB0</param>
    <param name="baudrate">115200</param>
    <param name="slaveID">80</param>

  </hardware>
  <sensor name="imu" type="imu_sensor">
    <state_interface name="orientation.x"/>
    <state_interface name="orientation.y"/>
    <state_interface name="orientation.z"/>
    <state_interface name="orientation.w"/>
    <state_interface name="angular_velocity.x"/>
    <state_interface name="angular_velocity.y"/>
    <state_interface name="angular_velocity.z"/>
    <state_interface name="linear_acceleration.x"/>
    <state_interface name="linear_acceleration.y"/>
    <state_interface name="linear_acceleration.z"/>
  </sensor>
    <sensor name="imu2" type="imu_sensor">
    <state_interface name="orientation.x"/>
    <state_interface name="orientation.y"/>
    <state_interface name="orientation.z"/>
    <state_interface name="orientation.w"/>
    <state_interface name="angular_velocity.x"/>
    <state_interface name="angular_velocity.y"/>
    <state_interface name="angular_velocity.z"/>
    <state_interface name="linear_acceleration.x"/>
    <state_interface name="linear_acceleration.y"/>
    <state_interface name="linear_acceleration.z"/>
  </sensor>

</ros2_control>

    <!-- control  -->
    <ros2_control name="diff_drive" type="system">
      <hardware>
        <!--control manager read xml file and tranfer param for harware interface (hardwareiterface::HardwareInfor info)  -->
        <plugin>diff_hardware_interface/DiffDrivewareInterface</plugin>
        <!-- param "left_wheel_name" has value "left_wheel_joint" -->
        <param name="left_wheel_name">left_wheel_joint</param>
        <param name="right_wheel_name">right_wheel_joint</param>
        <param name="idLeft">2</param>
        <param name="idRight">3</param>
        <param name="rWheel">0.075</param>
        <param name="widthRobot">0.48</param>
        <param name="canPort">slcan0</param>
      </hardware>
      <!-- joint name have same name in xacrofile same name(state_interface) to broadcaster joint correct  -->
      <joint name="left_wheel_joint"> 
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="right_wheel_joint">
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
    </ros2_control>  

</robot>


